components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterUtil.java

```diff
@@ -172,14 +173,20 @@ public static void makeAsyncAPICall(SecurityEventTokenPayload securityEventToken
             log.debug("Publishing event data to WebSubHub. URL: " + url + " tenant domain: " + tenantDomain);
         }
 
+        WebSubHubCorrelationLogUtils.triggerCorrelationLogForRequest(request);
+        long requestStartTime = System.currentTimeMillis();
```

Comment:
just a fyi, this way we can make sure the value doesn't get change
```suggestion
        final long requestStartTime = System.currentTimeMillis();
```